op-e2e/migration_test.go

```diff
@@ -0,0 +1,376 @@
+	config.l2Path = os.Getenv("OP_E2E_MIGRATION_L2_DATA_PATH")
+	if config.l2Path == "" {
+		panic("must specify an l2 data path")
+	}
+
+	migrationDataDir := path.Join(cwd, "..", "packages", "migration-data", "data")
+	config.ovmAddrsPath = path.Join(migrationDataDir, "ovm-addresses.json")
+	config.evmAddrsPath = path.Join(migrationDataDir, "evm-addresses.json")
+	config.ovmAllowancesPath = path.Join(migrationDataDir, "ovm-allowances.json")
+	config.ovmMessagesPath = path.Join(migrationDataDir, "ovm-messages.json")
+	config.evmMessagesPath = path.Join(migrationDataDir, "evm-messages.json")
+}
+
+type storageSlot struct {
+	addr  string
+	slot  string
+	value string
+}
+
+const (
+	networkName  = "mainnet-forked"
+	hardhatImage = "docker.io/ethereumoptimism/hardhat-node:latest"
+)
+
+var hardcodedSlots = []storageSlot{
+	{
+		"0xdE1FCfB0851916CA5101820A69b13a4E276bd81F",
+		"0x0",
+		"0x000000000000000000000000f39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+	},
+	{
+		"0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1",
+		"0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103",
+		"0x000000000000000000000000f39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+	},
+	{
+		"0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1",
+		"0x33",
+		"0x000000000000000000000000f39Fd6e51aad88F6F4ce6aB8827279cffFb92266",
+	},
+}
+
+func TestMigration(t *testing.T) {
+	if !config.enabled {
+		t.Skipf("skipping migration tests")
+		return
+	}
+
+	lgr := testlog.Logger(t, log.LvlDebug)
+	lgr.Info("starting forked L1")
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	dkr, err := client.NewClientWithOpts(client.FromEnv)
+	require.NoError(t, err, "error connecting to Docker")
+
+	//_, err = dkr.ImagePull(context.Background(), hardhatImage, types.ImagePullOptions{})
+	//require.NoError(t, err, "error pulling hardhat image")
+
+	realL1Client, err := ethclient.Dial(config.l1URL)
+	require.NoError(t, err)
+	headBlock, err := realL1Client.HeaderByNumber(ctx, nil)
+	require.NoError(t, err)
+	// Have to specify a small confirmation depth here to prevent the Hardhat fork
+	// from timing out in the middle of contract deployments.
+	forkBlock, err := realL1Client.BlockByNumber(ctx, new(big.Int).Sub(headBlock.Number, big.NewInt(10)))
+	require.NoError(t, err)
+	forkBlockNumber := forkBlock.NumberU64()
+
+	lgr.Info("writing deploy config")
+	deployCfg := e2eutils.ForkedDeployConfig(t, e2eutils.DefaultMnemonicConfig, forkBlock)
+	deployCfgPath := path.Join(cwd, "..", "packages", "contracts-bedrock", "deploy-config", "mainnet-forked.json")
+	f, err := os.OpenFile(deployCfgPath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0o744)
+	require.NoError(t, err)
+	enc := json.NewEncoder(f)
+	enc.SetIndent("", "  ")
+	require.NoError(t, enc.Encode(deployCfg))
+
+	ctnr, err := dkr.ContainerCreate(ctx, &container.Config{
+		Image: hardhatImage,
+		Env: []string{
+			fmt.Sprintf("FORK_STARTING_BLOCK=%d", forkBlockNumber),
+			fmt.Sprintf("FORK_URL=%s", config.l1URL),
+			"FORK_CHAIN_ID=1",
```

Comment:
Hmm I don't think we need the chain id to be set to 1, it makes me a bit worried about replay attacks with the transactions that are sent during our tests

Reply:
We do, otherwise transactions revert with an invalid chain ID. Replay attacks shouldn't matter because we're overwriting storage slots to use dev accounts. No real accounts initiate a transaction.